mapboxgl/viz.py

```diff
@@ -345,3 +360,88 @@ def add_unique_template_variables(self, options):
             lineWidth=self.line_width,
         ))
 
+
+class AltChoroplethViz(MapViz):
+    """Create a choropleth viz"""
+
+    def __init__(self,
+                 data,
+                 vector_polygon_source=False,
+                 vector_url=None,
+                 vector_layer_name=None,
+                 vector_join_color_property=None,
+                 join_data=[],
+                 data_join_property=None,
+                 label_property=None,
+                 color_property=None,
+                 color_stops=None,
+                 color_default='grey',
+                 color_function_type='interpolate',
+                 line_color='white',
+                 line_stroke='solid',
+                 line_width=1,
+                 *args,
+                 **kwargs):
+        """Construct a Mapviz object
+
+        :param label_property: property to use for marker label
+        :param color_property: property to determine circle color
+        :param color_stops: property to determine circle color
+        :param color_default: property to determine default circle color if match lookup fails
+        :param color_function_type: property to determine `type` used by Mapbox to assign color
+        :param line_color: property to determine choropleth line color
+        :param line_stroke: property to determine choropleth line stroke (solid, dashed, dotted, dash dot)
+        :param line_width: property to determine choropleth line width
+
+        """
+        super(AltChoroplethViz, self).__init__(data, *args, **kwargs)
+
+        self.template = 'choropleth_alt'
+        self.vector_polygon_source = vector_polygon_source
+        self.vector_url = vector_url
+        self.vector_layer_name = vector_layer_name
+        self.vector_join_color_property = vector_join_color_property
+        self.join_data = join_data
+        self.data_join_property = data_join_property
+        self.label_property = label_property
+        self.color_property = color_property
+        self.color_stops = color_stops
+        self.color_default = color_default
+        self.color_function_type = color_function_type
+        self.line_color = line_color
+        self.line_stroke = line_stroke
+        self.line_width = line_width
+
+    def add_unique_template_variables(self, options):
+        """Update map template variables specific to heatmap visual"""
+
+        # set line stroke dash interval based on line_stroke property
+        if self.line_stroke == "dashed":
```

Comment:
Nice.  Very nice for data viz use cases to just specify the type of viz they want instead of writing style json code or Mapbox expressions 👍